src/ConfigEditor.tsx

```diff
@@ -1,148 +1,170 @@
-import { InlineField, Input, Select } from '@grafana/ui';
-import { DataSourcePluginOptionsEditorProps, SelectableValue } from '@grafana/data';
-import { CentreonMetricOptions, EAccess, CentreonMetricSecureDatas } from './types';
-
-interface Props extends DataSourcePluginOptionsEditorProps<CentreonMetricOptions, CentreonMetricSecureDatas> {}
-
-interface State {}
-
-export class ConfigEditor extends PureComponent<Props, State> {
-  onChangeCentreonURL = (event: ChangeEvent<HTMLInputElement>) => {
-    const { onOptionsChange, options } = this.props;
+import React, { ChangeEvent } from 'react';
 
-    let centreonURL = event.target.value;
-
-    const jsonData = {
+import { InlineField, Input, Select } from '@grafana/ui';
+import {
+  DataSourcePluginOptionsEditorProps,
+  SelectableValue,
+} from '@grafana/data';
+
+import {
+  CentreonMetricOptions,
+  CentreonMetricSecureDatas,
+  EAccess,
+} from './@types/types';
+
+type Props = DataSourcePluginOptionsEditorProps<
+  CentreonMetricOptions,
+  CentreonMetricSecureDatas
+>;
+
+export const ConfigEditor = (props: Props): JSX.Element => {
+  const { onOptionsChange, options } = props;
+  const onChangeCentreonURL = (event: ChangeEvent<HTMLInputElement>): void => {
+    const centreonURL = event.target.value;
+
+    const newJSONData = {
       ...options.jsonData,
       centreonURL,
     };
     onOptionsChange({
       ...options,
-      jsonData: jsonData,
+      jsonData: newJSONData,
     });
   };
 
-  onChangeUsername = (event: ChangeEvent<HTMLInputElement>) => {
-    const { onOptionsChange, options } = this.props;
-
-    const jsonData = {
+  const onChangeUsername = (event: ChangeEvent<HTMLInputElement>): void => {
+    const newJSONData = {
       ...options.jsonData,
       username: event.target.value,
     };
     onOptionsChange({
       ...options,
-      jsonData: jsonData,
+      jsonData: newJSONData,
     });
   };
 
   // Secure field (only sent to the backend)
-  onChangePassword = (event: ChangeEvent<HTMLInputElement>) => {
-    const { onOptionsChange, options } = this.props;
-
+  const onChangePassword = (event: ChangeEvent<HTMLInputElement>): void => {
     const { value } = event.target;
 
-    let secureJsonData = options.secureJsonData || {};
-    let secureJsonFields = options.secureJsonFields;
-    let jsonData = options.jsonData;
+    const newSecureJsonData = options.secureJsonData || {};
+    const newSecureJsonFields = options.secureJsonFields;
+    const newJSONData = options.jsonData;
     if (options.jsonData.access === EAccess.PROXY) {
-      secureJsonData.password = value;
-      jsonData.password = '';
+      newSecureJsonData.password = value;
+      newJSONData.password = '';
     } else {
-      jsonData.password = value;
-      secureJsonFields.password = false;
-      secureJsonData.password = '';
+      newJSONData.password = value;
+      newSecureJsonFields.password = false;
+      newSecureJsonData.password = '';
```

Comment:
BTW, there is a function in Ramda that updates a property in an object: https://ramdajs.com/docs/#set